### DISCUSSION TITLE
StreamingResponse was block my program???

### DISCUSSION THREAD

**Discussion by noaft** [fastapi/fastapi#13024] [Questions] [answered]
### Privileged issue

- [X] I'm @tiangolo or he asked me directly to create an issue here.

### Issue Content

async def generate_predictions(model_name):
    global stop_flag
    mean, std, labels, model, pca_model = get_init(model_name)  
    async for prediction in collect_predict(mean, std, labels, model, pca_model):
        if stop_flag == True:
            stop_flag = False 
            break
        yield json.dumps({"prediction": prediction}) + "\n"

@router.get("/end_predict")
def end_predict():
    global stop_flag
    stop_flag = True
    return {'message' : 'End predict'}

@router.post("/predict")
def predict_func(data: predict):
    model_name = data.name_predict
    return StreamingResponse(
        generate_predictions(model_name), 
        media_type="application/json"
    )

**Comment by YuriiMotov** [comment, marked as answer]
Most likely the problem is in `collect_predict`. Make sure this function is not blocking

**Reply by noaft** [reply]
I was solution by create break time in stream function.
